pkg/processor/worker/allocator.go

```diff
@@ -83,6 +85,9 @@ func NewSingletonWorkerAllocator(parentLogger logger.Logger, worker *Worker) (Al
 }
 
 func (s *singleton) Allocate(time.Duration) (*Worker, error) {
+	if s.isTerminated {
+		return nil, ErrAllWorkersAreTerminated
```

Comment:
you now return an error, so we would want a different, more informative log in the callers instead of `Failed to allocate worker`

Reply:
@TomerShor it is not the same as `Failed to allocate worker`. Check below:

```
var ErrAllWorkersAreTerminated = errors.New("All workers are terminated")
```

Reply:
@rokatyy I know, but look at [this](https://github.com/nuclio/nuclio/blob/4405f11a6efbaf7b9e1136d1fd118e4d5ba8bf41/pkg/processor/trigger/kafka/trigger.go#L229-L234) for example, it does not make the differentiation between errors, and still fails with `failed ...`.

Reply:
@TomerShor But it's really a failure of worker allocation and wrapped error will contain a reason why worker hadn't been allocated. Why not?

Reply:
We need to check how the function monitor behaves in cases where the pod is deleted:
Does it mark the function as unhealthy before the pod restarts?
If there is an error in the function (such as like we're adding now) - will it mark the function as unhealthy?
After being unhealthy - does it return to `Ready` automatically?

Reply:
@TomerShor from what I see in the code, it depends on the number of replicas (if we had only one pod, then we will have unhealthy state until pod restart). However, if the available replicas are greater than or equal to minReplicas, we should be fine here. I believe that this is the expected behavior. What do you think?

Reply:
OK, leave it as is.
I guess the function pod will only be terminated unexpectedly if something did happen.